### ISSUE TITLE
Best practice: amp-selector: Unexpected focus handling / keyboard interaction

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31166]
## WCAG Level
Level A

## Priority
High

## Type
Navigation

## Pages/screens/Components affected
* [amp-selector](https://amp.dev/documentation/components/amp-selector/)

## Description
For people who navigate using a keyboard or keyboard-like device (such as a switch control or a joystick), the focus order of interactive controls in a document should be logical and intuitive.

In an `amp-selector` (which is exposed as an ARIA `role="listbox"`), currently all options are individually focusable (through the use of `tabindex="0"`). This does not conform to the expected way in which listboxes operate - generally, a listbox as a whole receives focus, and the user then navigates through its options using the cursor keys (see the [W3C WAI-ARIA Authoring Practices 1.1 Listbox keyboard interaction](https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction) section).

## User impact
While the `amp-selector` construct is still keyboard-operable (and thus does not normatively fail [WCAG 2.1 Success Criterion 2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG21/#focus-order)), users may initially have trouble understanding how to use the selector.

More problematically, when assistive technologies encounter a `role="listbox"`, they will assume that the listbox follows conventional listbox behaviour, which can cause problems for users. For instance, NVDA will switch into "forms mode", meaning that once the user is focused on one of the selector options, the user won't be able to use regular cursor reading keys to move through content (since NVDA assumes that cursor keys are needed to move between options) - to proceed as normal, users will have to explicitly exit forms mode. In the case of JAWS, users will hear the instruction "to move to an item press the arrow keys", which will currently have no effect. Overall, the experience will be confusing for these users.

## Required solution
Refer to the [W3C WAI-ARIA Authoring Practices 1.1 Listbox keyboard interaction](https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction) section. Instead of making all options explicitly focusable with `tabindex="0"`, only make a single option focusable, and handle cursor key interactions to change the currently focused option programmatically.

While no demo was provided, the documentation also suggests that `amp-selector` can contain other structures such as carousels. In this case, the resulting structure may not be suitable as a "listbox". In these cases, drop the `role="listbox"` / `role="option"` approach and instead simply treating the `amp-selector` as a generic `role="group"` with a collection of toggle buttons (see the [W3C WAI-ARIA Authoring Practices 1.1 Button](https://www.w3.org/TR/wai-aria-practices-1.1/#button) section) - these can be individually focusable, and have no additional keyboard interaction requirements (and thus won't lead to issues of "forms mode" in NVDA or incorrect instructions in JAWS).

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Turn on your screen reader.
2. Navigate to the `amp-selector` listbox.
3. Verify that the keyboard behaviour of the listbox matches the ARIA practices recommendation.

## Related standards
* [WCAG 2.1 Success Criterion 2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG21/#focus-order)

## More information
* [W3C WAI-ARIA Authoring Practices 1.1 Listbox keyboard interaction](https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction)

## Test data
Test date: November 2020
